Cython/Compiler/PyrexTypes.py

```diff
@@ -4404,6 +4410,110 @@ def error_condition(self, result_code):
         return "dummy"
 
 
+class PythonTypeConstructor(PyrexType):
```

Comment:
Is this a type? And, if so, is it a type that inherits straight from `PyrexType`? Not a Python object type?

Reply:
It should hopefully always end up specialized hence always end up transformed to another type. I'm not sure quite what an unspecialized `PythonTypeConstructor` means at the end of the code generation stage.

I guess it probably makes sense to fall back to looking like a `PyObjectType` though so I'll do this.